### PR TITLE
Fix component governance alerts

### PR DESCRIPTION
**Pull request by sebastienros**
###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/5164)

### CODE REVIEW

File: src/Components/Aspire.Microsoft.Data.SqlClient/Aspire.Microsoft.Data.SqlClient.csproj [hunk 1]
```diff
@@ -25,11 +25,16 @@
...
     <PackageReference Include="AspNetCore.HealthChecks.SqlServer" />
-    <PackageReference Include="Microsoft.Data.SqlClient" /> <!-- Force the version from Directory.Packages.props -->
     <PackageReference Include="Microsoft.Extensions.Configuration.Binder" />
     <PackageReference Include="Microsoft.Extensions.Diagnostics.HealthChecks" />
     <PackageReference Include="Microsoft.Extensions.Hosting.Abstractions" />
     <PackageReference Include="OpenTelemetry.Extensions.Hosting" />
+
+    <!-- Component Governance -->
+    <PackageReference Include="Microsoft.Data.SqlClient" />
+    <PackageReference Include="Azure.Identity" />
```
[the remark this answers is not in the document]
Comment (joperezr) [reply src/Components/Aspire.Microsoft.Data.SqlClient/Aspire.Microsoft.Data.SqlClient.csproj:35]: Same question for other projects where you didn't only add Azure.Core, which I think were like 3 of them or so

File: playground/orleans/OrleansClient/OrleansClient.csproj [hunk 2]
```diff
@@ -16,6 +16,9 @@
     <AspireProjectOrPackageReference Include="Aspire.Azure.Data.Tables" />
     <ProjectReference Include="..\OrleansContracts\OrleansContracts.csproj" />
     <ProjectReference Include="..\OrleansServiceDefaults\OrleansServiceDefaults.csproj" />
+
+    <!-- Component Governance -->
+    <PackageReference Include="Azure.Core" />
```
Comment (eerhardt) [inline playground/orleans/OrleansClient/OrleansClient.csproj:21]: Can we use this feature instead? https://learn.microsoft.com/en-us/nuget/consume-packages/Central-Package-Management#transitive-pinning

That way we don't need PackageReference in all these projects.

Comment (sebastienros) [reply playground/orleans/OrleansClient/OrleansClient.csproj:21]: Looks great, will do that.

**Comment by sebastienros** [comment]
Updated with transitive pinning.

File: Directory.Packages.props [hunk 3]
```diff
@@ -5,6 +5,7 @@
   <Import Project="eng/Versions.props" Condition="'$(MajorVersion)' == '' and Exists('eng/Versions.props')" />
   <PropertyGroup>
     <ManagePackageVersionsCentrally>true</ManagePackageVersionsCentrally>
+    <CentralPackageTransitivePinningEnabled>true</CentralPackageTransitivePinningEnabled>
```
Comment (joperezr) [inline Directory.Packages.props:8]: Nice! didn't know this was a thing.

Comment (joperezr) [reply Directory.Packages.props:8]: The only thing that makes me a bit worried is that this now may be pinning down transitive dependencies, but I guess that in that case we'd get NuGet warnings, right?

**Review by joperezr — APPROVED** [review]
Thanks for fixing, Seb.

File: Directory.Packages.props (hunk 3 above)
Comment (sebastienros) [reply Directory.Packages.props:8]: I am trying to test this, several Azure SDK libraries requires `Azure.Core >= 1.40.0` (only used as a transitive dependency) so I included `1.39.0` and got errors:

```
Severity	Code	Description	Project	File	Line	Suppression State	Details
Error (active)	NU1109	Detected package downgrade: Azure.Core from 1.40.0 to centrally defined 1.38.0. Update the centrally managed package version to a higher version. 
 DatabaseMigration.ApiModel -> Azure.Identity 1.12.0 -> Azure.Core (>= 1.40.0) 
 DatabaseMigration.ApiModel -> Azure.Core (>= 1.38.0)	DatabaseMigration.ApiModel	D:\aspire\playground\DatabaseMigration\DatabaseMigration.ApiModel\DatabaseMigration.ApiModel.csproj	1		
```

Comment (joperezr) [reply Directory.Packages.props:8]: Ok great, no concerns then.

**Pull request merged**